clients/drcachesim/drcachesim.dox.in

```diff
@@ -81,6 +82,86 @@ analysis tools (see \ref sec_drcachesim_newtool).
+copies of the binaries and reading the raw bytes for each instruction
+in order to obtain the opcode and full operand information.
+See also \ref sec_drcachesim_core.
+
+Branch targets are also not explicitly recorded (a design
+decision). The executed target as well as branch direction can be
+obtained by examining the address of the instruction immediately
+following a branch. If the program flow is changed by the kernel such
+as by signal delivery, the branch target is explicitly recorded in the
+trace in a metadata marker entry of type
+#TRACE_MARKER_TYPE_KERNEL_EVENT.
+
+\section sec_drcachesim_format_data Memory Access Records
+
+Memory accesses (data loads and stores) are stored in #_memref_data_t.
+The program counter of the instruction performing the memory access,
+the virtual address (unless \ref sec_drcachesim_phys are enabled), and
+the size are provided.
+
+\section sec_drcachesim_format_other Other Records
+
+Besides instruction and memory records, other trace entry types
+include #_memref_marker_t, #_memref_flush_t, #_memref_thread_exit_t,
+etc. These records provide specific inforamtion about events that can
+alter the program flow or the system's states.
+
+Trace markers are particularly important to allow reconstruction of
+the program execution.  Marker records in #_memref_marker_t provide
+metadata identifying some event that occurred at this point in the
+trace.  Each marker record contains two additional fields:
+
+- \p marker_type - identifies the type of marker
+- \p marker_value - carries the value of the marker
+
+Some of the more important markers are:
+
+- #TRACE_MARKER_TYPE_KERNEL_EVENT - This identifies kernel-initiated control transfers such as signal delivery.  The next instruction record is the start of the handler for a kernel-initiated event.  The value of this type of marker contains the program counter at the kernel interruption point.  If the interruption point is just after a branch, this value is the target of that branch.
+
+- #TRACE_MARKER_TYPE_KERNEL_XFER - This identifies a system call that changes control flow, such as a signal return..
+
+- #TRACE_MARKER_TYPE_TIMESTAMP - The marker value provides a timestamp for this point of the trace (in units of microseconds since Jan 1, 1601 UTC). This value can be used to synchronize records from different threads. It is used in the sequential analysis of a multi-threaded trace.
+
+- #TRACE_MARKER_TYPE_CPU_ID - The marker value contains the CPU identifier on which the subsequent records were collected. It is useful to help track thread migrations occurring during execution.
+
+- #TRACE_MARKER_TYPE_FUNC_ID, #TRACE_MARKER_TYPE_FUNC_RETADDR, #TRACE_MARKER_TYPE_FUNC_ARG, #TRACE_MARKER_TYPE_FUNC_RETVAL - These markers are used to capture information about function calls.  Which functions to capture must be explicitly selected at tracing time.  Typical candiates are heap allocation and freeing functions.  See sec_drcachesim_funcs.
```

Comment:
\ref sec_drcachesim_funcs`

Reply:
Done and confirmed it makes a proper link.